### PR TITLE
Added Node.js workflow.

### LINKED ISSUE #11: Node.js workflow? [improvement]
Since NuCat uses Node.js for both client and server side, it would make sense to have a workflow for it. 

Should someone (or me) add this?

### PR DESCRIPTION
**Pull request by AdvaitDhingra**
I have set up and tested the node.js workflow. I have modified it to work with NuCat. It tests both the client and the server side.

Shold fix [this](https://github.com/icecube/NuCat/issues/11) issue.

### CODE REVIEW

**Comment by AdvaitDhingra** [comment]
Took quite a few commits, but it works

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Test chnages (Node.js)
```
Comment (ChujieChen) [inline .github/workflows/node.js.yml]: Let's keep the name simple: 'NodeJS'

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Test Client Side
+      uses: actions/setup-node@v2
+      with:
+        node-version: 12.18.3
```
Comment (ChujieChen) [inline .github/workflows/node.js.yml]: How about: `12.x`

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - run: npm --prefix /client ci
+    - run: npm --prefix /client run build
+
+
+    - name: Test Server Side
+      uses: actions/setup-node@v2
+      with:
+        node-version: 12.18.3
```
Comment (ChujieChen) [inline .github/workflows/node.js.yml]: same here

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Test Client Side
+      uses: actions/setup-node@v2
+      with:
+        node-version: 12.18.3
+    - run: npm --prefix /client ci
+    - run: npm --prefix /client run build
```
Comment (ChujieChen) [inline .github/workflows/node.js.yml]: The above two lines did not work. They cause errors:
```
Run npm --prefix /client ci
npm --prefix /client ci
shell: /usr/bin/bash -e {0}
Error: EACCES: permission denied, mkdir '/client'
```

File: .github/workflows/node.js.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+
+    - name: Test Server Side
+      uses: actions/setup-node@v2
+      with:
+        node-version: 12.18.3
+    - run: npm --prefix /server ci
+    - run: npm --prefix /server run server
```
Comment (ChujieChen) [inline .github/workflows/node.js.yml]: Same here. The above two lines never worked.

**Review by ChujieChen — CHANGES_REQUESTED** [review]
Hey @AdvaitDhingra . Thank you for your code! This YMAL didn't work. Those four `npm run` lines never got a chance to run (please check the 'Details' page of this workflow). Please revise this file. Also please follow the `CONTRIBUTING.md` for future PRs.

**Comment by AdvaitDhingra** [comment]
Thanks for the review. I'll fix theses errors.

**Comment by ChujieChen** [comment]
Hi @AdvaitDhingra , the workflow never gets run. And thus, I'm closing this PR. Please follow the [CONTRIBUTING.md](https://github.com/icecube/NuCat/blob/main/CONTRIBUTING.md) for future PRs (try squashing your commits with rebase -i to avoid a long list of commits). Thank you!

**Pull request closed without merging**